pkg/bbgo/order_execution.go

```diff
@@ -321,30 +322,46 @@ func (c *BasicRiskController) ProcessOrders(
 
 type OrderCallback func(order types.Order)
 
+type batchOrderStatus struct {
+	Index int
+	Order *types.Order
+	Err   error
+}
+
 // BatchPlaceOrder
 func BatchPlaceOrder(
 	ctx context.Context, exchange types.Exchange, orderCallback OrderCallback, submitOrders ...types.SubmitOrder,
 ) (types.OrderSlice, []int, error) {
-	var createdOrders types.OrderSlice
-	var err error
+	g, _ := errgroup.WithContext(ctx)
 
-	var errIndexes []int
+	results := make([]batchOrderStatus, len(submitOrders))
 	for i, submitOrder := range submitOrders {
-		createdOrder, err2 := exchange.SubmitOrder(ctx, submitOrder)
-		if err2 != nil {
-			err = multierr.Append(err, err2)
-			errIndexes = append(errIndexes, i)
-		}
+		i, submitOrder := i, submitOrder
+		g.Go(func() error {
```

Comment:
I wonder if we can really benefits from this goroutine, cuz binance set the rate limiter 10 orders per second 

(Spot Order Rate Limit (for a single account) 100 orders per 10 seconds)

strategies like grid2 depends on this order execution, and the sequence of the orders are important.

would prefer a concurrent option for this, and apply the original behaivor by default.